src/v/kafka/server/handlers/alter_partition_reassignments.cc

```diff
@@ -64,59 +64,72 @@ partitions_request_iterator validate_replicas(
     return valid_range_end;
 }
 
+/*
+ * begin          - starting position for a vector<reassignable_partition>
+ * end            - stopping position for a vector<reassignable_partition>
+ * resp_it        - a wrapper to std::back_inserter
+ * topic_response - a reassignable_topic_response to put errors into
+ * alive_nodes    - list of RP nodes that are live
+ * tp_metadata    - topic metadata used to check replication factor
+ *
+ * return:  an iterator that represents the stop position of all valid
+ * partitions
+ */
 template<typename ResultIter>
 partitions_request_iterator validate_partitions(
   partitions_request_iterator begin,
   partitions_request_iterator end,
   ResultIter resp_it,
   reassignable_topic_response topic_response,
-  std::vector<model::node_id> all_node_ids) {
+  std::vector<model::node_id> alive_nodes,
+  std::optional<cluster::topic_metadata> tp_metadata) {
```

Comment:
If `resp_it` is supposed to be a `std::back_insert_iterator` why not make it as such?

```c++
template<typename Container>
partitions_request_iterator validate_partitions(
  partitions_request_iterator begin,
  partitions_request_iterator end,
  std::back_insert_iterator<Container> resp_it,
  reassignable_topic_response topic_response,
  std::vector<model::node_id> alive_nodes,
  std::optional<cluster::topic_metadata> tp_metadata) {
```